### DISCUSSION TITLE
[@lit-labs/ssr] with Astro: "elementRenderers is not iterable"

### DISCUSSION THREAD

**Discussion by davidenke** [lit/lit#3784] [Q&A] [answered]
Hi,

I'm using lit and Astro in a monorepo. I've a ui package containing my lit elements, which are bundled with esbuild excluding the lit runtime (`external: ['lit*']`).

In Astro I'm importing an element in an Astro Component to map the properties to the element. Using the Astro directive `client:load`, I'd try to do a ssr build.
```mdx
---
import { Button } from '@scope/ui';
---
<Button client:load label="Button"></Button>
```

But I'm constantly getting the error `TypeError: elementRenderers is not iterable at getElementRenderer` from the `@lit-labs/ssr` package.

To check if the elements in question are able to be build using ssr, I tested each element using the `@lit-labs/testing/ssr-fixture` with `@open-wc/testing` and aligned them accordingly.

I think I've a conceptual problem about how to use lit elements in Astro. Is anybody having a working example or does recognize the issue with my setup?

The following versions (right now latest) of my dependencies are:
```json
    "@astrojs/lit": "2.0.0",
    "@lit-labs/ssr": "3.1.1",
    "astro": "2.1.9",
    "lit": "2.7.1",
    "typescript": "5.0.3"
```

**Comment by augustjk** [comment, marked as answer]
Looks like there's an issue filed in the Astro repo for this: https://github.com/withastro/astro/issues/6743

I think I know where the problem is with the Astro lit plugin. I can add more details to the issue and work on a fix there.
